pkg/util/containers/v2/metrics/collector_containerd.go

```diff
@@ -0,0 +1,416 @@
+		return nil
+	}
+
+	var cpuLimits *specs.LinuxCPU
+	if OCISpec != nil && OCISpec.Linux != nil && OCISpec.Linux.Resources != nil {
+		cpuLimits = OCISpec.Linux.Resources.CPU
+	}
+
+	cpuLimitPct := float64(system.HostCPUCount())
+	if cpuLimits != nil && cpuLimits.Period != nil && *cpuLimits.Period > 0 && cpuLimits.Quota != nil && *cpuLimits.Quota > 0 {
+		cpuLimitPct = float64(*cpuLimits.Quota) / float64(*cpuLimits.Period)
+	}
+
+	limit := cpuLimitPct * timeDiff
+	return &limit
+}
+
+func getContainerdStatsWindows(windowsStats *wstats.WindowsContainerStatistics) *ContainerStats {
+	if windowsStats == nil {
+		return nil
+	}
+
+	return &ContainerStats{
+		Timestamp: windowsStats.Timestamp,
+		CPU:       getContainerdCPUStatsWindows(windowsStats.Processor),
+		Memory:    getContainerdMemoryStatsWindows(windowsStats.Memory),
+		IO:        getContainerdIOStatsWindows(windowsStats.Storage),
+	}
+}
+
+func getContainerdCPUStatsWindows(procStats *wstats.WindowsContainerProcessorStatistics) *ContainerCPUStats {
+	if procStats == nil {
+		return nil
+	}
+
+	return &ContainerCPUStats{
+		Total:  uint64ToFloat64Ptr(procStats.TotalRuntimeNS),
+		System: uint64ToFloat64Ptr(procStats.RuntimeKernelNS),
+		User:   uint64ToFloat64Ptr(procStats.RuntimeUserNS),
+	}
+}
+
+func getContainerdMemoryStatsWindows(memStats *wstats.WindowsContainerMemoryStatistics) *ContainerMemStats {
+	if memStats == nil {
+		return nil
+	}
+
+	return &ContainerMemStats{
+		PrivateWorkingSet: uint64ToFloat64Ptr(memStats.MemoryUsagePrivateWorkingSetBytes),
+		CommitBytes:       uint64ToFloat64Ptr(memStats.MemoryUsageCommitBytes),
+		CommitPeakBytes:   uint64ToFloat64Ptr(memStats.MemoryUsageCommitPeakBytes),
+	}
+}
+
+func getContainerdIOStatsWindows(ioStats *wstats.WindowsContainerStorageStatistics) *ContainerIOStats {
+	if ioStats == nil {
+		return nil
+	}
+
+	return &ContainerIOStats{
+		ReadBytes:       uint64ToFloat64Ptr(ioStats.ReadSizeBytes),
+		WriteBytes:      uint64ToFloat64Ptr(ioStats.WriteSizeBytes),
+		ReadOperations:  uint64ToFloat64Ptr(ioStats.ReadCountNormalized),
+		WriteOperations: uint64ToFloat64Ptr(ioStats.WriteCountNormalized),
+	}
+}
+
+func getContainerdNetworkStatsWindows() *ContainerNetworkStats {
+	// Network stats are not exposed on Windows, so return 0's for all the
+	// fields.
+
+	return &ContainerNetworkStats{
+		BytesSent:   util.Float64Ptr(0),
+		BytesRcvd:   util.Float64Ptr(0),
+		PacketsSent: util.Float64Ptr(0),
+		PacketsRcvd: util.Float64Ptr(0),
+		Interfaces:  make(map[string]InterfaceNetStats),
+	}
+}
+
+func getOpenFileDescriptors(processes []containerd.ProcessInfo) int {
+	fileDescCount := 0
+
+	for _, p := range processes {
+		pid := p.Pid
+		fdCount, err := providers.ContainerImpl().GetNumFileDescriptors(int(pid))
+		if err != nil {
+			log.Debugf("Failed to get file desc length for pid %d: %s", pid, err)
+			continue
+		}
+		fileDescCount += fdCount
+	}
+
+	return fileDescCount
+}
+
+func uint64ToFloat64Ptr(x uint64) *float64 {
```

Comment:
Instead of defining this here, reuse the helper function introduced in the ECS provider https://github.com/DataDog/datadog-agent/pull/9739

Reply:
it's merged 👍

Reply:
I've changed this collector to reuse that function :+1: